projects/packages/sync/src/class-main.php

```diff
@@ -25,7 +25,7 @@ public static function configure() {
 			add_action( 'plugins_loaded', array( __CLASS__, 'on_plugins_loaded_late' ), 90 );
 		}
 		// Any hooks below are special cases that need to be declared even if Sync is not allowed.
-		add_action( 'jetpack_user_authorized', array( 'Automattic\\Jetpack\\Sync\\Actions', 'do_initial_sync' ), 10, 0 );
```

Comment:
We may need to maintain the jetpack_user_authorized hook.

It fires on authorization of each user. Which ensures that Jetpack has the correct user roles for that user. As such we might want to have both jetpack_site_registered and jetpack_user_authorized.

If this is the case we alls need to keep that check in the sync _allowed

Reply:
> We may need to maintain the jetpack_user_authorized hook.

> It fires on authorization of each user. Which ensures that Jetpack has the correct user roles for that user. As such we might want to have both jetpack_site_registered and jetpack_user_authorized.

I took a deeper look at how those events are processed on WPCOM and I agree with you on that.

> If this is the case we alls need to keep that check in the sync _allowed

Nevertheless, I don't think we should modify the check in `sync_allowed`. The logic there is that sync relies on a site level connection to function. If we don't have that then sync should not be allowed. However, when a user connection occurs, a site level connection is given, so sync will be allowed anyway.

Therefore, my suggestion would be to keep the `jetpack_user_authorized` hook, but not as an exception in `Main` but on `Users` instead, where it seems to better fit.

Suggestion on Users `init`:

```
public static function init() {
        add_action( 'jetpack_user_authorized', array( 'Automattic\\Jetpack\\Sync\\Actions', 'do_initial_sync' ), 10, 0 );
	$connection = new Jetpack_Connection();
	if ( $connection->has_connected_user() ) {
		// Kick off synchronization of user role when it changes.
		add_action( 'set_user_role', array( __CLASS__, 'user_role_change' ) );
	}
}
```

As always your thoughts are very appreciated.

Reply:
Thanks for adding jetpack_user_authorized back in and placement in the Users module is 💯